### PR TITLE
Upgrade PyJWT to v1.7.1 (#2069)

### PR DESCRIPTION
**Pull request by natanlao**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `HCA` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] Added `chain` label <sub>or this PR is not the base of another PR</sub>
- [x] Made this PR a blocker of next PR in chain <sub>or this PR is not the base of another PR</sub>
- [x] Ran `make update_requirements` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or commented that sandbox will be skipped</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit

Primary reviewer (after pushing merge commit)

- [ ] Moved linked issue to Merged
- [ ] Pushed merge commit to Github
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Unassign reviewer from PR

### CODE REVIEW

File: UPGRADING.rst [hunk 1]
```diff
@@ -11,6 +11,12 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
 have too many entries in this file.
 
 
+#2069 Upgrade PyJWT to >=1.7.1
```
Comment (hannes-ucsc) [inline UPGRADING.rst:14]:
```suggestion
#2069 Upgrade PyJWT to 1.7.1
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/5892

You checked the box but did you actually run `make requirements_update`?

**Comment by natanlao** [comment]
@hannes-ucsc 

> You checked the box but did you actually run make requirements_update?

[Yes.](https://github.com/DataBiosphere/azul/blob/1d8f604aaa816a81f18613b88544dd50c1c95c03/requirements.dev.trans.txt) Upon investigation, I believe that this discrepancy occurred because:
* google-api-core@1.22.2 was [released](https://pypi.org/project/google-api-core/#history) on September 3
* cfn-lint@0.35.1 was [released](https://pypi.org/project/cfn-lint/#history) on September 2

both of which were cut after I filed this PR. (Unless I am misunderstanding how `make requirements_update` works.)

I'll update this PR after running `make requirements_update` again.

**Comment by hannes-ucsc** [comment]
> both of which were cut after I filed this PR

… but before the most recent review. It's not sufficient to just do this once. Some PRs take weeks to get approved. Treat every review cycle as if it were the first.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**